app/modules/handler/containers/Prompt.js

```diff
@@ -53,12 +54,18 @@ class PromptContainer extends Component<Props> {
     actions.signURI(tx, blockchain, wallet);
   }
   templateURI = () => {
-    const { blockchains, prompt, wallets } = this.props;
-    const { chainId } = prompt;
+    const { blockchains, prompt: oldprompt, wallets } = this.props;
+    const { chainId } = oldprompt;
     // Set the blockchain for this network
     const blockchain = find(blockchains, { chainId });
     // Find the default wallet for this chain (defaults to first at the moment)
-    const defaultWallet = find(wallets, { chainId });
+
+    const specifiedActor = get(prompt, 'req.1.authorization.0.actor');
+    const specifiedPermission = get(prompt, 'req.1.authorization.0.permission');
+    const defaultWallet =
+      find(wallets, { chainId, account: specifiedActor, authorization: specifiedPermission }) ||
```

Comment:
It'd be cleaner if you just named the variables `account` and `authorization`.

```
    const account = get(prompt, 'req.1.authorization.0.actor');
    const authorization = get(prompt, 'req.1.authorization.0.permission');
    const defaultWallet =
      find(wallets, { chainId, account, authorization }) ||
```

Reply:
good point!